rust/src/meas/randomized_response/mod.tex

```diff
@@ -0,0 +1,250 @@
+
+\section{Algorithm Implementation}
+\subsection{Code in Rust}
+
+The current OpenDP library contains the \texttt{make\_randomized\_response\_bool} function implementing the binary randomized response measurement. This is defined in lines 28-53 of the file \texttt{mod.rs} in the Git repository \url{https://github.com/opendp/opendp/blob/main/rust/src/meas/randomized_response/mod.rs#L28-L53}
+
+In \texttt{make\_randomized\_response\_bool}, which accepts a parameter \texttt{prob} of type \texttt{Q} and a parameter \texttt{constant\_time} of type \texttt{bool}, the function takes in a boolean data point \texttt{arg} and returns the truthful value \texttt{arg} with probability \texttt{prob} and the untruthful value $\texttt{!arg}$ with probability $1 - \texttt{prob}$ (in constant time, if the flag is turned on). 
+
+\subsection{Pseudo Code in Python}
+We present a simplified Python-like pseudocode of the Rust implementation below. The necessary definitions for the pseudocode can be found at \href{https://github.com/opendp/whitepapers/blob/pseudocode-defns/pseudocode-defns/pseudocode_defns.pdf}{``List of definitions used in the pseudocode"}. % need to fix
+
+\emph{The use of \texttt{code}-style parameters in the preconditions section below (for example, \texttt{input\_domain}) means that this information should be passed along to the \texttt{Measurements} constructor.}
+
+\subsubsection*{Preconditions}
+To ensure the correctness of the output, we require the following preconditions:
+
+\begin{itemize}
+    \item \textbf{User-specified types:}
+    \begin{itemize}
+        \item Variable \texttt{prob} must be of type \texttt{Q}
+        \item Variable \texttt{constant\_time} must be of type \texttt{bool}
+        \item Type \texttt{bool} must have trait \texttt{SampleBernoulli<Q>}
+        \item Type \texttt{Q} must have traits \texttt{float}, \texttt{ExactIntCast<IntDistance>},\\ \texttt{DistanceConstant<IntDistance>}, \texttt{InfSub}, \texttt{InfLn}
+        \item Variable \texttt{IntDistance} must have trait \texttt{InfCast<Q>}
+    \end{itemize}
+\end{itemize}
+
+\subsubsection*{Postconditions}
+\begin{itemize}
+    \item A \texttt{Measurement} is returned (i.e., if a \texttt{Measurement} cannot be returned successfully, then an error should be returned).
+\end{itemize}
+
+
+\begin{lstlisting}[language=Python, escapechar=|]
+def make_randomized_response_bool(prob : Q, constant_time : bool):
+    input_domain = AllDomain(bool)
+    output_domain = AllDomain(bool)
+    input_metric = DiscreteMetric()
+    similarity_metric = MaxDivergence()
+    
+    if (prob < 0.5 or prob >= 1): |\label{line:range}|
+        raise Exception("probability must be in [0.5, 1)")
+
+    c = inf_ln(inf_div(prob, neg_inf_sub(1, prob)))
+    def privacy_map(d_in: u_32) -> u_32: |\label{line:map}|
+        return d_in * c;
```

Comment:
is there implicit casting here?  Also, I think it would be nicer to do this as "return 0 if d_in = 0, else return c" (see also my suggested edit to def of DiscreteMetric in the pseudocode defs doc)